backend/migration/utils.py

```diff
@@ -0,0 +1,131 @@
+        dest_client=dest_client, 
+        source_collection_name=source_collection_name, 
+        destination_collection_name=destination_collection_name,
+        batch_size=batch_size
+    )
+
+
+def _has_custom_shards(source_client: QdrantBase, collection_name: str) -> bool:
+    collection_info = source_client.get_collection(collection_name)
+    return (
+        getattr(collection_info.config.params, "sharding_method", None)
+        == models.ShardingMethod.CUSTOM
+    )
+
+def _recreate_collection(
+    source_client: QdrantBase,
+    dest_client: QdrantBase,
+    source_collection_name: str,
+    destination_collection_name: str,
+
+) -> None:
+    src_collection_info = source_client.get_collection(source_collection_name)
+    src_config = src_collection_info.config
+    src_payload_schema = src_collection_info.payload_schema
+
+    dest_client.recreate_collection(
+        destination_collection_name,
+        vectors_config=src_config.params.vectors,
+        sparse_vectors_config=src_config.params.sparse_vectors,
+        shard_number=src_config.params.shard_number,
+        replication_factor=src_config.params.replication_factor,
+        write_consistency_factor=src_config.params.write_consistency_factor,
+        on_disk_payload=src_config.params.on_disk_payload,
+        hnsw_config=models.HnswConfigDiff(**to_dict(src_config.hnsw_config)),
+        optimizers_config=models.OptimizersConfigDiff(**to_dict(src_config.optimizer_config)),
+        wal_config=models.WalConfigDiff(**to_dict(src_config.wal_config)),
+        quantization_config=src_config.quantization_config,
+    )
+
+    _recreate_payload_schema(dest_client, destination_collection_name, src_payload_schema)
+
+
+def _recreate_payload_schema(
+    dest_client: QdrantBase,
+    collection_name: str,
+    payload_schema: Dict[str, models.PayloadIndexInfo],
+) -> None:
+    for field_name, field_info in payload_schema.items():
+        dest_client.create_payload_index(
+            collection_name,
+            field_name=field_name,
+            field_schema=field_info.data_type if field_info.params is None else field_info.params,
+        )
+
+
+def _migrate_collection(
+    source_client: QdrantBase,
+    dest_client: QdrantBase,
+    source_collection_name: str,
+    destination_collection_name: str,
+    batch_size: int = 100,
+) -> None:
+    """Migrate collection from source client to destination client
+
+    Args:
+        source_collection_name (str): Collection name of source
+        destination_collection_name (str): Collection name of destination
+        source_client (QdrantBase): Source client
+        dest_client (QdrantBase): Destination client
+        batch_size (int, optional): Batch size for scrolling and uploading vectors. Defaults to 100.
+    """
+    records, next_offset = source_client.scroll(source_collection_name, limit=2, with_vectors=True)
+    dest_client.upload_points(destination_collection_name, records, wait=True)  # type: ignore
```

Comment:
odd, why is this needed? specifically `limit=2` ?

Reply:
This is just to get the initial offset. It can be any number I have just kept it as a small number. To iterate over other points later if any.